Sources/MuxUploadSDK/Upload/ChunkedFileUploader.swift

```diff
@@ -240,9 +243,26 @@ fileprivate actor Worker {
         try chunkedFile.seekTo(byte: startingReadCount)
         
         let startTime = Date().timeIntervalSince1970
-        
-        let fileSize = chunkedFile.fileSize
-        let wideFileSize = Int64(fileSize)
+
+        let fileAttributes = try FileManager.default.attributesOfItem(
+            atPath: uploadInfo.videoFile.path
+        )
+        guard let fileSize = fileAttributes[
+            FileAttributeKey.size
+        ] as? UInt64 else {
+            // TODO: Does the worker need its own error defintion?
+            throw ChunkedFileError.invalidState("Cannot retrieve file size")
+        }
+
```

Comment:
Can you discuss the benefits of unwrapping the boilerplate?

Reply:
Sure, in the previous implementation there was a possibility an inaccurate file size could be read from `ChunkedFile` if its `fileSize` property was not set. That introduces the potential for issues elsewhere in the SDK. The current implementation retrieves the file size from a single source of truth.

Reply:
I'm totally on board with querying the file manager every time, I'm just wondering why take out the helper function? Working with this system API requires handling 3 distinct error cases, and standard Java/Kotlin practice would be to wrap that up and provide 1 case. 

Is it more idiomatic in swift to just deal with each case every time?

Reply:
What helper function are you referring to? Do you mean the one I added to `ChunkedFile` in an earlier commit and then removed?

Reply:
(I squashed my changes so some of the in-between states disappeared, but just want to make sure I didn't accidentally remove something that was already in the SDK outside of this PR.)

Reply:
That's what I was referring to. Like I said, it's not a blocker. There's just not really many java/android system APIs that expose things like file size as a key in a dict. If it's common in swift to just deal with all the error-checking/validation every single time you use the API, then I guess we wouldn't need a method.

This would be a departure from our Java/Kotlin code style for sure, though

Reply:
> That's what I was referring to. Like I said, it's not a blocker. There's just not really many java/android system APIs that expose things like file size as a key in a dict. If it's common in swift to just deal with all the error-checking/validation every single time you use the API to get file size, then I guess we wouldn't need a method.
> 
> This would be a departure from our Java/Kotlin code style for sure, though

I see, that's a good point re adding a helper function. The case of `FileManager` is odd because `attributesOfItem(atPath:)` is a translated Objective-C method, the need to cast from `Any` to `UInt64` isn't very Swifty. [Foundation does have a type safe extension on NSDictionary for the file size.](https://developer.apple.com/documentation/foundation/nsdictionary/1413465-filesize/)
One quick win here is to switch over to use and remove the need to handle the failure case when the type cast from `Any` fails (`as? UInt64`).

More generally: my understanding is that the convention in Swift is to propagate thrown errors unless there is (1) a need to wrap them to e.g. hide implementation details or (2) to provide an additional context that wouldn't otherwise be present.

Since methods that use the file size are internal to the SDK as are their direct callers, hiding implementation details seems unnecessary. And the only extra context added by `throw ChunkedFileError.invalidState("Cannot retrieve file size")` is that we're trying to fetch the file size, that can also be easily inferred due to the fact that the SDK doesn't use or fetch other attributes. 

So rather than wrapping, would do this when in a throwing method:

```
try fileManager.attributesOfItem(atPath:videoFile.path).fileSize()
```

Wdyt?

Reply:
Ok this did not come as clean as I'd hoped, Swift compiler isn't cooperating. Here's what this looks like instead, the `as NSDictionary` is the only way the compiler will permit calling `fileSize()`.

```
(try FileManager.default.attributesOfItem(atPath: uploadInfo.videoFile.path) as NSDictionary).fileSize()
```

I'll make this cleaner by adding an extension method to `FileManager` that calls this.